lib/autoupdate/agent/installer_test.go

```diff
@@ -420,9 +447,235 @@ func TestLocalInstaller_Link(t *testing.T) {
+	t.Parallel()
+	const version = "new-version"
+	servicePath := filepath.Join(serviceDir, serviceName)
+
+	tests := []struct {
+		name            string
+		installDirs     []string
+		installFiles    []string
+		installFileMode os.FileMode
+		existingLinks   []string
+		existingFiles   []string
+
+		resultPaths []string
+		errMatch    string
+	}{
+		{
+			name: "present with new links",
+			installDirs: []string{
+				"bin",
+				"bin/somedir",
+				"etc",
+				"etc/systemd",
+				"etc/systemd/somedir",
+				"somedir",
+			},
+			installFiles: []string{
+				"bin/teleport",
+				"bin/tsh",
+				"bin/tbot",
+				servicePath,
+				"README",
+			},
+			installFileMode: os.ModePerm,
+
+			resultPaths: []string{
+				"bin/teleport",
+				"bin/tsh",
+				"bin/tbot",
+				"lib/systemd/system/teleport.service",
+			},
+		},
+		{
+			name: "present with non-executable files",
+			installDirs: []string{
+				"bin",
+				"bin/somedir",
+				"etc",
+				"etc/systemd",
+				"etc/systemd/somedir",
+				"somedir",
+			},
+			installFiles: []string{
+				"bin/teleport",
+				"bin/tsh",
+				"bin/tbot",
+				servicePath,
+				"README",
+			},
+			installFileMode: 0644,
+
+			errMatch: "executable",
+		},
+		{
+			name: "present with existing links",
+			installDirs: []string{
+				"bin",
+				"bin/somedir",
+				"etc",
+				"etc/systemd",
+				"etc/systemd/somedir",
+				"somedir",
+			},
+			installFiles: []string{
+				"bin/teleport",
+				"bin/tsh",
+				"bin/tbot",
+				servicePath,
+				"README",
+			},
+			installFileMode: os.ModePerm,
+			existingLinks: []string{
+				"bin/teleport",
+				"bin/tsh",
+				"bin/tbot",
+			},
+			existingFiles: []string{
+				"lib/systemd/system/teleport.service",
+			},
+
+			errMatch: "refusing",
+		},
+		{
+			name: "conflicting systemd files",
+			installDirs: []string{
+				"bin",
+				"bin/somedir",
+				"etc",
+				"etc/systemd",
+				"etc/systemd/somedir",
+				"somedir",
+			},
+			installFiles: []string{
+				"bin/teleport",
+				"bin/tsh",
+				"bin/tbot",
+				servicePath,
+				"README",
+			},
+			installFileMode: os.ModePerm,
+			existingLinks: []string{
+				"lib/systemd/system/teleport.service",
+			},
+
+			errMatch: "replace irregular file",
+		},
+		{
+			name: "conflicting bin files",
+			installDirs: []string{
+				"bin",
+				"bin/somedir",
+				"etc",
+				"etc/systemd",
+				"etc/systemd/somedir",
+				"somedir",
+			},
+			installFiles: []string{
+				"bin/teleport",
+				"bin/tsh",
+				"bin/tbot",
+				servicePath,
+				"README",
+			},
+			installFileMode: os.ModePerm,
+			existingFiles: []string{
+				"bin/tsh",
+			},
+
+			errMatch: "replace file",
+		},
+		{
+			name:         "no links",
+			installFiles: []string{"README"},
+			installDirs:  []string{"bin"},
+
+			errMatch: "no binaries",
+		},
+		{
+			name:         "no bin directory",
+			installFiles: []string{"README"},
+
+			errMatch: "binary directory",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
```

Comment:
```suggestion
```

This shouldn't be necessary, because
1. Your subtests aren't parallel
2. Go 1.23 changes the behavior here anyway

Reply:
Thanks, removed from the whole package